src/components/PopupWithForm.js

```diff
@@ -6,8 +6,28 @@ export default class PopupWithForm extends Popup {
     this.submit = submitFunction;
     this.form = this.modal.querySelector(".form");
     this.inputs = this.form.querySelectorAll(".form__input");
+    this.button = this.form.querySelector(".form__submit-button");
+    this.buttonText = this.button.textContent;
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      if (this.form.id === "form-delete-card") {
+        this.button.textContent = "Deleting...";
+      } else {
+        this.button.textContent = "Saving...";
+      }
+    } else {
+      this.button.textContent = this.buttonText;
+    }
+  }
+
+  setUpDeleteModal = (cardObj) => {
```

Comment:
Needs correction:

1) For handling the confirmation you should create a separate class PopupWithConfirmation
2) `PopupWithConfirmation` is a general class. It can be used for any confirmation, not only card delete confirmation. This class should have a setter for `_handleFormSubmit`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

this._form.addEventListener("submit", (evt) => { 
  evt.preventDefault(); 
  this._handleFormSubmit(); 
})
```

Reply:
Still needs correction

Reply:
Sorry I didn't see these last time; I'm sure reviewing the same mistakes twice is annoying. I think I missed the expand comments button github has in the middle of the page. I had a question on if its it bad practice to set popup.handleFormSubmit directly outside the class vs doing it a function like you did?

Reply:
Setting it directly is the bad practice, because that way you wan't see what's going on with your class instance, it will be hard for understanding and debugging, so you should use the class methods for setting the variables and handlers (https://playcode.io/javascript/getter-setter)